### PR TITLE
Kbd state indicator

### PR DESCRIPTION
**Pull request by zjes**
After discussion in zjes/lxqt-kbdstate#6 this "third" party plugin merged to kdindicator plugin. I hope it will be usefull.

Fixes lxde/lxqt#602.
Fixes zjes/lxqt-kbdstate#6.

### CODE REVIEW

**Comment by paulolieuthier** [comment]
@jleclanche with this, KGuiAddons won't be a dependency anymore :smile:. xkbcommon will.

**Comment by paulolieuthier** [comment]
@agaida please, test.

File: plugin-kbindicator/CMakeLists.txt [hunk 1]
```diff
@@ -1,24 +1,66 @@
...
 
 set(UIS
-    lxqtkbindicatorconfiguration.ui
+    src/kbdstateconfig.ui
 )
 
 set(LIBRARIES
-    ${LIBRARIES}
-    KF5::GuiAddons
+    ${LXQT_LIBRARIES}
```
Comment (paulolieuthier) [inline plugin-kbindicator/CMakeLists.txt]: It should be `${LIBRARIES}` here.

**Comment by jleclanche** [comment]
@paulolieuthier Excellent.

**Comment by luis-pereira** [comment]
@paulolieuthier With the targets stuff, it should be ```lxqt```

**Comment by zjes** [comment]
So... 
Are any problems to solve before to merge it? :)

**Comment by palinek** [comment]
Just tried to compile...
- Should the build fail in case "XkbCommon X11 was found" ?

**Comment by zjes** [comment]
apt install libxkbcommon-x11-dev
Yes, needs more informative warning :)

**Comment by palinek** [comment]
> apt install libxkbcommon-x11-dev
> Yes, needs more informative warning :)

I know (you also need the libxkbcommon-dev because the cmake doesn't find xkbcommon-x11 w/o any information)... but what is the intended behaviour when there are no such libraries? (should the plugin compile/run, but only indicate num/caps/scroll-lock?)

Presentation issue - panel is vertical and the letters do not fit into the cell (in case there the iconSize is reasonably big... the information could be on multiple lines...):
screenshot


I also had to change the includes on current master to be able to compile:
```diff
diff --git a/plugin-kbindicator/kbindicator-plugin.cpp b/plugin-kbindicator/kbindicator-plugin.cpp
index b43de64..c81ba00 100644
--- a/plugin-kbindicator/kbindicator-plugin.cpp
+++ b/plugin-kbindicator/kbindicator-plugin.cpp
@@ -1,6 +1,6 @@
 #include <QObject>
 #include "src/kbdstate.h"
-#include "ilxqtpanelplugin.h"
+#include "../panel/ilxqtpanelplugin.h"

 class LxQtKbIndicatorPlugin: public QObject, public ILxQtPanelPluginLibrary
 {
diff --git a/plugin-kbindicator/src/kbdstate.h b/plugin-kbindicator/src/kbdstate.h
index 9a4070e..f958eed 100644
--- a/plugin-kbindicator/src/kbdstate.h
+++ b/plugin-kbindicator/src/kbdstate.h
@@ -1,7 +1,7 @@
 #ifndef _KDBSTATE_H_
 #define _KDBSTATE_H_

-#include "ilxqtpanelplugin.h"
+#include "../panel/ilxqtpanelplugin.h"
 #include "settings.h"
 #include "content.h"
 #include "kbdwatcher.h"
```

**Comment by zjes** [comment]
Thanks for this issues. :)

> there are no such libraries
Yes. Without X11 can be builded and can show only leds, without xkbcommon - plugin cannot be build at whole.

**Comment by agaida** [comment]
are there any risks if i try a field test in siduction? @zjes @palinek @paulolieuthier

**Comment by palinek** [comment]
@zjes for the cmake and build fail issue... IMO the `CMakeLists.txt` should be changed this way:
```diff
--- a/plugin-kbindicator/CMakeLists.txt
+++ b/plugin-kbindicator/CMakeLists.txt
@@ -30,11 +30,10 @@ set(LIBRARIES
 )
 
 find_package(PkgConfig REQUIRED)
+pkg_check_modules(XKB_COMMON REQUIRED xkbcommon)
 pkg_check_modules(XKB_COMMON_X11 QUIET xkbcommon-x11)
 
-if(NOT XKB_COMMON_X11_FOUND)
-    message(STATUS "XkbCommon X11 was not found")
-else()
+if(XKB_COMMON_X11_FOUND)
     message(STATUS "XkbCommon X11 was found")
     find_package(Qt5 COMPONENTS X11Extras Xml)
     pkg_check_modules(XCB_XCB xcb-xkb)
@@ -57,6 +56,8 @@ else()
     )
 
     add_definitions(-DX11_ENABLED)
+else()
+    message(FATAL_ERROR "No XkbCommon backend(X11) found!")
 endif()
 
 BUILD_LXQT_PLUGIN(${PLUGIN})
```

**Comment by paulolieuthier** [comment]
GTM after applying the patches.

**Comment by zjes** [comment]
Hi, tonight i will apply those patches. 
What this means - GTM? :)

**Comment by palinek** [comment]
> What this means - GTM? :)

good to merge

**Comment by zjes** [comment]
Patches were applied, vertical layout was "fixed" (not good enough, better suitable hmm... maybe some kind of flow layout?).

**Comment by paulolieuthier** [comment]
@zjes it's good enough. Now, please squash all the commits into one with a message like this:
```
kbindicator: depend on xkbcommon & show lang flag

That removes dependency on KGuiAddons. xkbcommon is light, stable,
installed everywhere and supports Wayland.
```

**Comment by zjes** [comment]
This is a little bit git magic for me... check this commit, all correct? :)

**Comment by paulolieuthier** [comment]
Squashing was perfect (it is, indeed, magic). Rebase is missing. But I can do it for you.

**Comment by paulolieuthier** [comment]
Merged in 3b9c7b9d5b07333047ca4620f8a441d0ec043771. @zjes, thanks again!

**Comment by zjes** [comment]
Thank you.  :)

**Pull request closed without merging**